src/main/java/org/ihtsdo/buildcloud/core/service/InputFileServiceImpl.java

```diff
@@ -315,7 +301,51 @@ private void gatherSourceFilesFromExternallyMaintainedBucket(String centerKey, S
 	@Override
 	public InputStream getSourceFileStream(String releaseCenterKey, String productKey, String source, String sourceFileName) {
 		Product product = constructProduct(releaseCenterKey, productKey);
-		return fileHelper.getFileStream(s3PathHelper.getProductSourcesPath(product) + source + BuildS3PathHelper.SEPARATOR + sourceFileName);
+		return fileHelper.getFileStream(s3PathHelper.getProductSourcesPath(product) + source + S3PathHelper.SEPARATOR + sourceFileName);
 	}
 
+	@Override
+	public void copyExternallyMaintainedFiles(String releaseCenterKey, String source, String target) {
+		String sourcePath = s3PathHelper.getExternallyMaintainedDirectoryPath(releaseCenterKey, source);
+		String targetPath = s3PathHelper.getExternallyMaintainedDirectoryPath(releaseCenterKey, target);
+
+		List<String> externalFiles = fileHelper.listFiles(sourcePath);
+		for (String externalFile : externalFiles) {
+			// Skip if current object is a directory
+			if (StringUtils.isBlank(externalFile) || externalFile.endsWith(S3PathHelper.SEPARATOR)) {
+				continue;
+			}
+			try {
+				String sourceFilePath = sourcePath + externalFile;
+				String targetFilePath = targetPath + externalFile.replaceAll(source, target);
+
+				File tmpFile = File.createTempFile("sct2-file", ".txt");
+
+				try {
+					InputStream inputStream = fileHelper.getFileStream(sourceFilePath);
+					try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+						 PrintWriter writer = new PrintWriter(new BufferedOutputStream(new FileOutputStream(tmpFile)))) {
+						String str;
+						boolean copyHeaderComplete = false;
+						while ((str = reader.readLine()) != null) {
+							if (str != null && !copyHeaderComplete) {
+								writer.println(str);
+								copyHeaderComplete = true; // should not we break out of while here?
```

Comment:
@CoderMChu This method is moved here from `ReleaseServiceImpl`.

Not quite sure if we should not break out of the loop once `copyHeaderComplete = true`. Probably I am missing something here, but thought I'll add a comment. Will remove this comment before merging this PR.

Reply:
@astro-snail It looks like only the header line is copied. Yes it should break out. This method is added for FRI-148 to prepare files for the new authoring cycle. Therefore it would be wrong to use this somewhere else without refactoring. We could add another parameter for example isHeaderOnly so that it can be reused when all data is needed.